src/vorta/views/archive_tab.py

```diff
@@ -365,3 +368,28 @@ def row_of_archive(self, archive_name):
         items = self.archiveTable.findItems(archive_name, QtCore.Qt.MatchExactly)
         rows = [item.row() for item in items if item.column() == 4]
         return rows[0] if rows else None
+
+    def delete_action(self):
+        params = BorgDeleteThread.prepare(self.profile())
+        if not params['ok']:
+            self._set_status(translate(params['message']))
+            return
+
+        archive_name = self.selected_archive_name()
+        if archive_name is not None:
+            params['cmd'][-1] += f'::{archive_name}'
+
+            thread = BorgDeleteThread(params['cmd'], params, parent=self)
+            thread.updated.connect(self._set_status)
+            thread.result.connect(self.delete_result)
+            self._toggle_all_buttons(False)
+            thread.start()
+        else:
+            self._set_status("No archive selected")
+
+    def delete_result(self, result):
+        if result['returncode'] == 0:
+            self._set_status(self.tr('Archive deleted.'))
+            self.list_action()
```

Comment:
In theory we could just remove the archive from the local DB is the delete-command was successful without listing/syncing all archives.

On the other hand it saves some code and syncing archives is always a good idea.